File: java/PracticaPDOO/src/main/java/deepspace/Loot.java
/**
 *
 * @author Alejandro Villanueva Prados
 */
package deepspace;

public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    Loot(int supplies, int weapons, int shields, int hangars, int medals) {
        nSupplies = supplies;
        nWeapons = weapons;
        nShields = shields;
        nHangars = hangars;
        nMedals = medals;
    }
    
    int getNSupplies() {
        return nSupplies;
    }
    
    int getNWeapons() {
        return nWeapons;
    }
    
    int getNShields() {
        return nShields;
    }
    
    int getNHangars() {
        return nHangars;
    }
    
    int getNMedals() {
        return nMedals;
    }
}

File: ruby/Loot.rb
require_relative "LootToUI"

module Deepspace
  class Loot
    def initialize(sup, weap, shields, hangars, medals, ef=false, city=false)
      @nSupplies = sup
      @nWeapons = weap
      @nShields = shields
      @nHangars = hangars
      @nMedals = medals
      @efficient = ef
      @spaceCity = city
    end

    def self.newCopy(l)
      l.clone
    end

    attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals, :efficient, :spaceCity

    def to_s
      "Suministros: #{nSupplies} - Armas: #{nWeapons} - Escudos: #{nShields} - Hangares: #{nHangars} - Medallas: #{nMedals} - Efficient: #{efficient} - Space City: #{spaceCity}"
    end

    def getNSupplies
      @nSupplies
    end

    def getNWeapons
      @nWeapons
    end

    def getNShields
      @nShields
    end

    def getNHangars
      @nHangars
    end

    def getNMedals
      @nMedals
    end

    def getUIversion
      LootToUI.new(self)
    end

    def clone  # Overrided to do deep copy instead
      Loot.new(nSupplies, nWeapons, nShields, nHangars, nMedals, efficient, spaceCity)
    end
  end
end
